slacker/slack_api.py

```diff
@@ -43,3 +48,32 @@ def post(self, method, args = {}):
       raise SlackAPIException('Unsuccessful API request: {}\nError: {}'.format(response.url, error))
 
     return data
+
+  def download_file(self, file_id, folder):
+    """Download file via ID to a folder. File IDs can be retrieved using the `files.list'
+    command. Private files use Bearer authorization via the token."""
+    if not os.path.exists(folder):
+      os.makedirs(folder, exists_ok = True)
+
+    headers = {}
+    file_info = self.post('files.info', {'file': file_id})['file']
+    if file_info['is_public']:
+      url = file_info['url_download']
+    else:
+      url = file_info['url_private_download']
+      headers['Authorization'] = 'Bearer {}'.format(self.__token)
+
+    self.__logger.debug('Downloading {} to {}'.format(url, folder))
+
+    res = requests.get(url, stream = True, headers = headers)
+    if res.status_code != 200:
+      self.__logger.warning('Unable to download {}'.format(url))
```

Comment:
Can we also log the status code and message from the response here?

Reply:
Good point, I'll try to make it similar to `post`.